pyexchange/kucoin.py

```diff
@@ -0,0 +1,269 @@
+        return self.price
+
+    @property
+    def remaining_buy_amount(self) -> Wad:
+        return self.amount*self.price if self.is_sell else self.amount
+
+    @property
+    def remaining_sell_amount(self) -> Wad:
+        return self.amount if self.is_sell else self.amount*self.price
+
+    def __repr__(self):
+        return pformat(vars(self))
+
+    @staticmethod
+    def from_list(item: list, pair: str, sell: bool):
+        return Order(order_id=item[5],
+                     pair=pair,
+                     is_sell=sell,
+                     price=Wad.from_number(item[2]),
+                     amount=Wad.from_number(item[3]))
+
+
+class Trade:
+    def __init__(self,
+                 trade_id: Optional[id],
+                 order_id: Optional[str],
+                 timestamp: float,
+                 pair: str,
+                 is_sell: bool,
+                 price: Wad,
+                 amount: Wad):
+        assert(isinstance(trade_id, int) or (trade_id is None))
+        assert(isinstance(timestamp, float))
+        assert(isinstance(pair, str))
+        assert(isinstance(is_sell, bool))
+        assert(isinstance(price, Wad))
+        assert(isinstance(amount, Wad))
+        assert(isinstance(order_id, str) or (order_id is None))
+
+        self.trade_id = trade_id
+        self.order_id = order_id
+        self.timestamp = timestamp
+        self.pair = pair
+        self.is_sell = is_sell
+        self.price = price
+        self.amount = amount
+
+    def __eq__(self, other):
+        assert(isinstance(other, Trade))
+        return self.trade_id == other.trade_id and \
+               self.order_id == other.order_id and \
+               self.timestamp == other.timestamp and \
+               self.pair == other.pair and \
+               self.is_sell == other.is_sell and \
+               self.price == other.price and \
+               self.amount == other.amount
+
+    def __hash__(self):
+        return hash((self.trade_id,
+                     self.order_id,
+                     self.timestamp,
+                     self.pair,
+                     self.is_sell,
+                     self.price,
+                     self.amount))
+
+    def __repr__(self):
+        return pformat(vars(self))
+
+    @staticmethod
+    def from_dict(pair, trade):
+        return Trade(trade_id=trade['id'],
+                     order_id=trade['orderOid'],
+                     timestamp=float(trade['createdAt']),
+                     pair=pair,
+                     is_sell=trade['direction'] == 'SELL',
+                     price=Wad.from_number(trade['dealPrice']),
+                     amount=Wad.from_number(trade['amount']))
+
+    @staticmethod
+    def from_list(pair, trade):
+        # [1544564526000, 'SELL', 25.0005, 0.0614088, 1.5352507, '5c102f2d335e7e08134edd77']
+        return Trade(trade_id=None,
+                     order_id=trade[5],
+                     timestamp=float(trade[0]),
```

Comment:
Didn't we use `int`s for timestamps for other exchanges..?

Reply:
change in 2nd commit